nrf_wifi/hw_if/hal/src/hal_api.c

```diff
@@ -1353,6 +1386,34 @@ void nrf_wifi_hal_dev_deinit(struct nrf_wifi_hal_dev_ctx *hal_dev_ctx)
 }
 
 
+enum nrf_wifi_status nrf_wifi_hal_lock_rx(struct nrf_wifi_hal_dev_ctx *hal_dev_ctx)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	unsigned long flags = 0;
+
+	nrf_wifi_osal_spinlock_irq_take(hal_dev_ctx->hpriv->opriv,
+					hal_dev_ctx->lock_rx,
+					&flags);
+
+	status = NRF_WIFI_STATUS_SUCCESS;
+
+	return status;
+}
+
+enum nrf_wifi_status nrf_wifi_hal_unlock_rx(struct nrf_wifi_hal_dev_ctx *hal_dev_ctx)
+{
+	enum nrf_wifi_status status = NRF_WIFI_STATUS_FAIL;
+	unsigned long flags = 0;
+
+	nrf_wifi_osal_spinlock_irq_rel(hal_dev_ctx->hpriv->opriv,
+				       hal_dev_ctx->lock_rx,
+				       &flags);
+
+	status = NRF_WIFI_STATUS_SUCCESS;
+
+	return status;
+}
+
```

Comment:
Do we actually need the above two functions (since they are just doing irq take and rel?

Reply:
This is the main fix in this PR, so, yes, we do need them, how else can FMAC access HAL's internal `lock_rx`?